core/trino-main/src/main/java/io/trino/likematcher/LikePatternMatcher.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.likematcher;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public record LikePatternMatcher(String pattern, Optional<Character> escape, LikeMatcher likeMatcher)
+{
+    public LikePatternMatcher(String pattern, Optional<Character> escape, LikeMatcher likeMatcher)
+    {
+        this.pattern = requireNonNull(pattern, "pattern is null");
+        this.escape = requireNonNull(escape, "escape is null");
+        this.likeMatcher = requireNonNull(likeMatcher, "likeMatcher is null");
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        LikePatternMatcher that = (LikePatternMatcher) o;
+        return Objects.equals(pattern, that.pattern) && Objects.equals(escape, that.escape);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(pattern, escape);
+    }
+
+    @Override
+    public String toString()
```

Comment:
Is this to string necessary?

Reply:
makes it easier to see what is in the compiled classes cache in debugger